File: frontend/src/types/enquiry.ts
export type PreferredMode = 'ONLINE' | 'IN_PERSON' | 'BOTH';
export type EnquiryStatus = 'NEW' | 'CONTACTED' | 'CONVERTED' | 'CLOSED';

export type Enquiry = {
  id: string;
  tutorSlug: string;
  parentName: string;
  parentEmail: string;
  parentPhone?: string;
  studentYear?: string;
  subject?: string;
  message?: string;
  preferredLocation?: string;
  preferredMode?: PreferredMode;
  status: EnquiryStatus;
  createdAt: string;
};

export type EnquiryPayload = Omit<Enquiry, 'id' | 'tutorSlug' | 'status' | 'createdAt'>;

File: backend/src/main/java/com/tutr/api/entity/Enquiry.java
package com.tutr.api.entity;

import com.tutr.api.converter.*;
import com.tutr.api.dto.*;
import com.tutr.api.entity.*;
import com.tutr.api.enums.*;
import com.tutr.api.repository.*;
import com.tutr.api.service.*;

import com.tutr.api.entity.BaseEntity;
import com.tutr.api.entity.TutorProfile;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "enquiries")
public class Enquiry extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tutor_profile_id", nullable = false)
    private TutorProfile tutorProfile;

    @Column(nullable = false)
    private String parentName;

    @Column(nullable = false)
    private String parentEmail;

    private String parentPhone;
    private String studentYear;
    private String subject;
    private String message;
    private String preferredLocation;

    @Enumerated(EnumType.STRING)
    private PreferredMode preferredMode;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EnquiryStatus status = EnquiryStatus.NEW;
}

